### PR TITLE
feat: Introduce table resize plan for scaling.

### PR DESCRIPTION
**Pull request by shanicky**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

first part of #13954

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

**Review by yezizp2012 — APPROVED** [review]
LGTM, let's see whether if using sorted pu in scheduler can fix the ci issue.

File: proto/meta.proto [hunk 1]
```diff
@@ -461,6 +462,22 @@ message RescheduleResponse {
...
+    uint32 parallelism = 1;
+  }
+
+  message AutoParallelism {}
+
+  message CustomParallelism {}
+
+  oneof parallelism {
+    FixedParallelism fixed = 1;
+    AutoParallelism auto = 2;
```
Comment (BugenZhao) [inline proto/meta.proto:476]: Would you mind adding some comments to describe them? For example, the respective behavior when auto scaling is on/off.

File: src/meta/src/stream/scale.rs [hunk 2]
```diff
@@ -2247,3 +2580,221 @@ impl GlobalStreamManager {
         (join_handle, shutdown_tx)
     }
 }
+
+// We redistribute parallel units (which will be ensembles in the future) through a simple consistent hashing ring.
+// Note that we have added some simple logic here to ensure the consistency of the ratio between each slot,
+// especially when equal division is needed.
+pub fn schedule_units_for_slots(
```
Comment (BugenZhao) [inline src/meta/src/stream/scale.rs:2587]: What about extracting this utility into separate files?

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Pull request merged**